### PR TITLE
#172 add error handling in components

### PR DESCRIPTION
**Pull request by bejow**
## Related Ticket/Issue
#172 

## Description
This PR adds:
- More effective error handling
- Changed Error state
- New Error Layout

### CODE REVIEW

**Comment by ludwigschubi** [comment]
@bejow Why is the diff for this pr so large?

**Comment by bejow** [comment]
@ludwigschubi because I had to branch out from app overview list. I mentioned it in the PR, that the other should be done before, so that there only the changes of this branch

File: src/utils/helper.js [hunk 1]
```diff
@@ -10,4 +10,29 @@ const isCmdPressed = (event) => {
...
+};
+
+const convertArrayToString = (array) => {
+    if (!Array.isArray(array)) {
+        return undefined;
+    }
+    let arrayString = '';
+    for (let i = 0; i < array.length; i++) {
+        arrayString += array[i] + '\n';
+    }
```
Comment (ludwigschubi) [inline src/utils/helper.js]: you can do this with the standard built-in method .join('\n') of the array

File: src/functional_components/ContactProfilePage/ContactProfilePage.js [hunk 2]
```diff
@@ -20,7 +24,12 @@ const ContactProfilePage = ({
     removeContact,
     webId,
     isContact,
+    error,
 }) => {
+    const errors = getErrorsFromErrorState(error);
+    if (errors.length > 0) {
+        throw new Error(convertArrayToString(errors));
+    }
```
[the remark this answers is not in the document]
Comment (ludwigschubi) [reply src/functional_components/ContactProfilePage/ContactProfilePage.js]: @bejow I saw you added the helper function, but you didn't use it inside of the components. pls do before we merge this

**Pull request merged**